src/headless/converse-chat.js

```diff
@@ -1110,8 +1110,11 @@ converse.plugins.add('converse-chat', {
                     return;
                 }
                 if (utils.isNewMessage(message) && this.isHidden()) {
-                    this.setFirstUnreadMsgId(message);
+                    if (this.get('num_unread') == 0) {
+                        this.setFirstUnreadMsgId(message);
+                    }
                     this.save({'num_unread': this.get('num_unread') + 1});
+                    this.sendMarker(message.get('from'), message.get('msgid'), 'displayed');
```

Comment:
Looks like you need to first check for the `is_markable` boolean saved on the message object before sending out a marker. We don't want to send out markers to messages that didn't ask for them.

Reply:
I have added the check for is_markable. That had an impact on the tests, so I I had to add markable to the generated text chat message from mock.createChatMessage